applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -0,0 +1,232 @@
+	char *ip;
+	/* Mobile Country Code*/
+	char *mccmnc;
+	/** Flag signifying that the data entry is to be encoded. */
+	bool queued;
+};
+
+struct cloud_data_ui {
+	/** Button number. */
+	int btn;
+	/** Button data timestamp. UNIX milliseconds. */
+	int64_t btn_ts;
+	/** Flag signifying that the data entry is to be encoded. */
+	bool queued;
+};
+
+struct cloud_codec_data {
+	/** Encoded output. */
+	char *buf;
+	/** Length of encoded output. */
+	size_t len;
+};
+
+static inline void cloud_codec_init(void)
+{
+	cJSON_Init();
+}
+
+int cloud_codec_decode_config(char *input, struct cloud_data_cfg *cfg);
+
+int cloud_codec_encode_config(struct cloud_codec_data *output,
+			      struct cloud_data_cfg *cfg);
+
+int cloud_codec_encode_data(struct cloud_codec_data *output,
+			    struct cloud_data_gps *gps_buf,
+			    struct cloud_data_sensors *sensor_buf,
+			    struct cloud_data_modem_static *modem_stat_buf,
+			    struct cloud_data_modem_dynamic *modem_dyn_buf,
+			    struct cloud_data_ui *ui_buf,
+			    struct cloud_data_accelerometer *accel_buf,
+			    struct cloud_data_battery *bat_buf);
+
+int cloud_codec_encode_ui_data(struct cloud_codec_data *output,
+			       struct cloud_data_ui *ui_buf);
+
+int cloud_codec_encode_batch_data(
+				struct cloud_codec_data *output,
+				struct cloud_data_gps *gps_buf,
+				struct cloud_data_sensors *sensor_buf,
+				struct cloud_data_modem_dynamic *modem_dyn_buf,
+				struct cloud_data_ui *ui_buf,
+				struct cloud_data_accelerometer *accel_buf,
+				struct cloud_data_battery *bat_buf,
+				size_t gps_buf_count,
+				size_t sensor_buf_count,
+				size_t modem_dyn_buf_count,
+				size_t ui_buf_count,
+				size_t accel_buf_count,
+				size_t bat_buf_count);
+
+void cloud_codec_populate_sensor_buffer(
+				struct cloud_data_sensors *sensor_buffer,
+				struct cloud_data_sensors *new_sensor_data,
+				int *head_sensor_buf,
+				size_t buffer_count);
+
+void cloud_codec_populate_ui_buffer(struct cloud_data_ui *ui_buffer,
+				    struct cloud_data_ui *new_ui_data,
+				    int *head_ui_buf,
+				    size_t buffer_count);
+
+void cloud_codec_populate_accel_buffer(
+				struct cloud_data_accelerometer *accel_buf,
+				struct cloud_data_accelerometer *new_accel_data,
+				int *head_accel_buf,
+				size_t buffer_count);
+
+void cloud_codec_populate_bat_buffer(struct cloud_data_battery *bat_buffer,
+				     struct cloud_data_battery *new_bat_data,
+				     int *head_bat_buf,
+				     size_t buffer_count);
+
+void cloud_codec_populate_gps_buffer(struct cloud_data_gps *gps_buffer,
+				     struct cloud_data_gps *new_gps_data,
+				     int *head_gps_buf,
+				     size_t buffer_count);
+
+void cloud_codec_populate_modem_dynamic_buffer(
+				struct cloud_data_modem_dynamic *modem_buffer,
+				struct cloud_data_modem_dynamic *new_modem_data,
+				int *head_modem_buf,
+				size_t buffer_count);
+
+static inline void cloud_codec_release_data(struct cloud_codec_data *output)
+{
+	k_free(output->buf);
```

Comment:
There's a dedicated `cJSON_FreeString()` function to release buffers allocated by CJSON lib.